### ISSUE TITLE
Resource allocation

### ISSUE THREAD

**Issue by ehgh** [HuguesTHOMAS/KPConv#17]
Hi,
I am trying to use your model on a dataset with various size frames. I faced two main issues:
- Is there an easy way to use multiple GPUs to train using your code? (The parallelism of training on different batches and averaging gradients on multiple GPUs at the same time)
- The model crashes on random epochs in larger models due to shortage of GPU memory allocation (I assume this is because frames have different sizes and hence, the batches will not be fixed size). Is there a way to have the model skip those batches during training to avoid crashes?
Thanks a lot.

**Comment by HuguesTHOMAS** [comment]
Hi @ehgh,

For your first question, multi GPU training is not easy to implement with Tensorflow, and not useful for our experiments, so we do not have the possibility to use it in our code. 

For your second question, our model is designed to work with various size frames. As explained in the paper, one batch is composed of a varying number of frames (depending on their size). It is filled with frames until a certain limit of points is reached. Therefore, the overall size of the batch should not vary much during training. 

The batch size parameters controls the average number of frames in a batch. Before the network starts, we compute the maximum number of points in a batch so that the average number of frames corresponds to this value. You should not set this parameter lower than 3 or 4 if you want GPU memory allocation to be stable.

If you look at the batch generator, the model randomly drops some points for the largest frames that cannot even fit in a batch. Therefore, the model should not crashed if you calibrated your batch limit well.
